### PR TITLE
Add support for custom options and custom params in customHelpers

### PR DESCRIPTION
**Pull request by san650**

### CODE REVIEW

**Comment by san650** [comment]
@juanazam please review!

File: DOCUMENTATION.md [hunk 1]
```diff
@@ -614,92 +614,135 @@ var page = PO.build({
...
-```
-
-If we execute:
 
-`page.users(3).userName()`
+assert.ok(page.userName().hasError(), 'user name has errors');
+```
 
-It would look for `tr:nth-of-type(3) input:first` in our page.
+As you can see the returned plan object is converted to a component.
```
Comment (juanazam) [inline DOCUMENTATION.md]: plan -> plain

File: test-support/page-object/custom-helper.js [hunk 2]
```diff
@@ -1,25 +1,31 @@
...
     return {
       buildPageObjectAttribute: function(key, parent) {
-        // def
-        return function() {
-          // eval
-          let qualifiedSelector = qualifySelector(parent.scope, selector),
-            response = userDefinedFunction(qualifiedSelector, options),
-            processedResponse;
+        return function(...params) {
+          let qualifiedSelector = qualifySelectorWithOptions(parent.scope, options.scope, options.index, selector),
```
Comment (juanazam) [inline test-support/page-object/custom-helper.js]: Can we split this into more lines? :smile:

**Comment by juanazam** [comment]
@san650 this is great! I left two comments, take a look at them and once we agree on them, we can merge this in :+1:

**Pull request merged**